utilities/mtv_migration.py

```
@@ -542,3 +542,74 @@ def verify_vm_disk_count(destination_provider, plan, target_namespace):
+                        raise AssertionError(
+                            f"Plan {plan.name} reached {status} before all plans were executing simultaneously. "
+                            f"Statuses: {status_msg}"
+                        )
+
+            # Check if all plans are executing
+            if all(plans_executing.values()):
+                LOGGER.info("SUCCESS: All migrations are executing simultaneously")
+                return
+
+    except TimeoutExpiredError:
+        raise AssertionError("Failed to validate all migrations executing simultaneously within timeout")
```

Ruff (0.14.14)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)